routes/user.js

```diff
@@ -0,0 +1,187 @@
+const router = require("express").Router();
+const bcrypt = require("bcryptjs");
+const Error = require("../models/Error");
+const Joi = require("@hapi/joi");
+const jwtService = require("../services/JwtService");
+
+const userService = require("../services/UserService");
+
+router
+    .route("/")
+    /*
+     * GET USER INFO
+     * [Only loggedIn user can see their info. The userID of loggedIn user is stored in the JWT token itself]
+     */
+    .get(async (req, res) => {
+        try {
+            const { userID } = jwtService.verify(req.headers["authorization"]);
+            if (userID == undefined) {
+                res.status(403).send(new Error("FORBIDDEN", "Unauthorized"));
+            } else {
+                const user = await userService.getUser(userID);
+                res.status(200).json({
+                    code: "OK",
+                    result: "SUCCESS",
+                    user: {
+                        zprn: user.zprn,
+                        firstName: user.firstName,
+                        lastName: user.lastName,
+                        department: user.department,
+                    },
+                });
+            }
+        } catch (error) {
+            res.status(500).send(new Error("INTERNAL_SERVER_ERROR", error));
+        }
+    })
+    /*
+     * ADD NEW USER
+     */
+    .post(async (req, res) => {
+        let {
+            zprn,
+            firstName,
+            lastName,
+            password,
+            department,
+            role,
+        } = req.body;
+        try {
+            const { error } = validateData(
+                zprn,
+                firstName,
+                lastName,
+                password,
+                department,
+                role
+            );
+            if (error !== undefined) {
+                res.status(401).send(
+                    new Error("BAD_REQUEST", error.details[0].message)
+                );
+            } else {
+                const salt = bcrypt.genSaltSync(10);
+                const hashedPassword = bcrypt.hashSync(password, salt);
+                await userService.addUser(
+                    zprn,
+                    firstName,
+                    lastName,
+                    hashedPassword,
+                    department,
+                    role
+                );
+                res.status(201).json({
+                    code: "CREATED",
+                    msg: "User added!",
+                    result: "SUCCESS",
+                });
+            }
+        } catch (error) {
+            res.status(500).send(new Error("INTERNAL_SERVER_ERROR", error));
+        }
+    })
+    /*
+     * UPDATE USER
+     * [This route only updates the user password]
+     */
+    .patch(async (req, res) => {
```

Comment:
Similarly here we need something like `/update` ?

Reply:
As discussed, only thing that can be changed about the user is the password, which is implemented here. The user can update any other details?

Reply:
Yes ofc, in the future we will be having the whole details of user updated. How can a system be possible if we cannot update the user details. We will be needing separate endpoint for this.

Reply:
Okay if we go by your way, tell me how will a authority update the details of a user? Or even his password?

Reply:
Okay, I'll create a route for updating the  user, while keeping the current route as well.
The current route will be used for changing the details, and new route for authority. I propose keeping both routes as I suppose users don't need to save his details (like userID) on their device.What say?

Reply:
You need to have a parameter for userid. We are developing an api and it needs to be as free and flexible as possible. We cannot constrain the client for doing things. We surely will be passing userid everywhere, ofc it's a major key in many operations. Surely userid will always be sent to the client.

Reply:
We'll provide a seperate route for the same. How's that?